### PR TITLE
fix scale down nodes ClOUD mode

### PR DESCRIPTION
**Pull request by fanux**

### CODE REVIEW

File: apply/cloud_applier.go [hunk 1]
```diff
@@ -35,13 +35,42 @@ func NewAliCloudProvider(cluster *v1.Cluster) Interface {
 	return &CloudApplier{d}
 }
 
+func (c *CloudApplier) ScaleDownNodes(cluster *v1.Cluster) (isScaleDown bool, err error) {
```
Comment (justadogistaken) [inline apply/cloud_applier.go:38]: This is a ScaleDownNodes func, but it update c.NodesToJoin within the func, feel confused

File: apply/current_cluster.go [hunk 2]
```diff
@@ -45,6 +54,8 @@ func getCurrentNodes() (*v1.Cluster, error) {
 		}
 		cluster.Spec.Nodes.IPList = append(cluster.Spec.Nodes.IPList, addr)
 	}
```
Comment (justadogistaken) [inline apply/current_cluster.go:56]: Feel curious about "Count", why we need to keep this value? just len(...) is fine?

File: apply/cloud_applier.go [hunk 3]
```diff
@@ -35,13 +35,42 @@ func NewAliCloudProvider(cluster *v1.Cluster) Interface {
...
+	logger.Info("desired master %s, current master %s, desired nodes %s, current nodes %s", c.ClusterDesired.Spec.Masters.Count,
+		cluster.Spec.Masters.Count,
+		c.ClusterDesired.Spec.Nodes.Count,
+		cluster.Spec.Nodes.Count)
+	if c.ClusterDesired.Spec.Masters.Count >= cluster.Spec.Masters.Count &&
```
Comment (justadogistaken) [inline apply/cloud_applier.go:43]: the invoking chain is too long, would it be better to declare the values at the beginning? just a suggestion

File: apply/cloud_applier.go (hunk 1 above)
Comment (fanux) [reply apply/cloud_applier.go:38]: Oh, we needn't update it, I will fixed it

File: apply/current_cluster.go (hunk 2 above)
Comment (fanux) [reply apply/current_cluster.go:56]: User will edit this value in CLOUD mod

File: apply/cloud_applier.go (hunk 3 above)
Comment (fanux) [reply apply/cloud_applier.go:43]: I think it's OKay.

**Review by kakaZhou719 — APPROVED** [review]

File: apply/cloud_applier.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (fanux) [reply apply/cloud_applier.go:38]: infra and baremetal Apply will check it.

**Pull request merged**